# Descover trade tendency, identify key trading partners, analyze specific product categories of Georgia in January - October 2023

In [1]:
import pandas as pd
import matplotlib.ticker as ticker
import plotly.express as px
import plotly.graph_objects as go
import numpy as np

pd.options.display.float_format = '{:.2f}'.format
pd.set_option('display.max_columns', None)
pd.set_option('display.max_columns', 200)
pd.set_option('display.max_rows', 100)
pd.set_option('display.min_rows', 100)
pd.set_option('display.expand_frame_repr', True)

In [2]:
# plot variables
label_size=14
title_size=22
fmt = ticker.EngFormatter(places=0)

# Data Collection
Data is collected from the Ministry of Finance of Georgia's public reports (https://www.mof.ge)

In [3]:
# Georgia's export-import 2023(except December)
jan_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_01.csv',dtype={"საქონლის კოდი":"object"})
feb_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_02.csv', dtype={"საქონლის კოდი":"object"})
mar_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_03.csv', dtype={"საქონლის კოდი":"object"})
apr_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_04.csv', dtype={"საქონლის კოდი":"object"})
may_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_05.csv', dtype={"საქონლის კოდი":"object"})
jun_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_06.csv', dtype={"საქონლის კოდი":"object"})
jul_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_07.csv', dtype={"საქონლის კოდი":"object"})
aug_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_08.csv', dtype={"საქონლის კოდი":"object"})
sep_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_09.csv', dtype={"საქონლის კოდი":"object"})
oct_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_10.csv', dtype={"საქონლის კოდი":"object"})
nov_23=pd.read_csv('/kaggle/input/geo-trade-2023/2023_11.csv', dtype={"საქონლის კოდი":"object"})

# Georgia's export-import 2022
df_2022 = pd.read_csv('/kaggle/input/geo-trade-2023/trade_2022.csv')

# HS Code names
codes = pd.read_csv('/kaggle/input/others/codes.csv', dtype={"Category":"object"})

# country names geo-english (original data is in Georgian)
countries = pd.read_csv('/kaggle/input/others/countries.csv')

In [4]:
may_23.head()

საქონლის კოდი   რეჟიმი  რაოდენობა  რაოდენობა კგ  ღირებულება დოლარი  \
0   90064000000  იმპორტი      60.00         50.00            4131.52   
1   85182995000  იმპორტი       1.00          0.23               2.01   
2   07070005000  იმპორტი  235255.00     235255.00           93466.43   
3   61045200000  იმპორტი      79.00          6.72             631.50   
4   83099090000  იმპორტი     287.00        287.00            3739.62   

  პარტნიორი ქვეყანა  
0           თურქეთი  
1          ყაზახეთი  
2           თურქეთი  
3        პორტუგალია  
4         ბულგარეთი

In [5]:
codes.head()

Category                                          Name
0     2603                  Copper ores and concentrates
1     3102  Mineral or chemical fertilisers, nitrogenous
2     7202                                  Ferro-alloys
3     2201                Waters (natural or artificial)
4     2202                                Mineral Waters

In [6]:
countries.head()

Partner      Country
0    ავსტრალია    Australia
1      ავსტრია      Austria
2     ავღანეთი  Afghanistan
3  აზერბაიჯანი   Azerbaijan
4     ალბანეთი      Albania

### Export - Import Dataframe of 2022

In [7]:
df_2022.head()

Country    Month     Export     Import
0    Albania  January     343.80   54768.44
1  Argentina  January       0.00 2020521.38
2    Armenia  January 9556544.60 2551888.00
3  Australia  January   66092.26       0.00
4    Austria  January  132138.24 3066550.94

In [8]:
df_2022.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1341 entries, 0 to 1340
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Country  1341 non-null   object 
 1   Month    1341 non-null   object 
 2   Export   1341 non-null   float64
 3   Import   1341 non-null   float64
dtypes: float64(2), object(2)
memory usage: 42.0+ KB


In [9]:
df_2022.describe()

Export       Import
count      1341.00      1341.00
mean    2898947.62   7533908.01
std     9591119.82  24060337.23
min           0.00         0.00
25%         200.00     49280.00
50%      116600.00    751365.51
75%      959051.49   4598264.17
max   120472702.69 213865581.15

# Data Cleaning and Transformation

In [10]:
data = [jan_23, feb_23,mar_23,apr_23,may_23,jun_23,jul_23,aug_23,sep_23,oct_23,nov_23]
months = ['January', 'February', 'March','April','May', 'June', 'July', 'August','September','October','November','December']

# Change column names for each dataframe and add month columns.
for df, month in zip(data, months):
    df.rename(columns={'საქონლის კოდი':'Code',
                           'რეჟიმი':'Type',
                           'რაოდენობა':'Quantity',
                           'რაოდენობა კგ':'Quantity_kg',
                           'ღირებულება დოლარი':'Cost',
                           'პარტნიორი ქვეყანა':'Partner'
                          }, inplace=True)
#     add month column.
    df['Month']=month
    
# Combine dataframes for each month into a single one. 
df_2023 = pd.concat(data,axis=0, sort=False, ignore_index=True)

# Add year columns to 2022 and 2023 Dataframes
df_2022['Year'] = '2022'
df_2023['Year'] = '2023'

df_2023.head()


Code     Type  Quantity  Quantity_kg      Cost     Partner    Month  \
0  70051030000  იმპორტი  22466.59    222546.67  33782.52       ირანი  January   
1  39219090000  იმპორტი  58782.40     58782.40 547996.80     თურქეთი  January   
2  39241000000  იმპორტი  10330.41     10330.41  33773.50     თურქეთი  January   
3  94016900000  იმპორტი     50.00        50.00     64.04  საფრანგეთი  January   
4  83021000009  იმპორტი   1150.84      1150.84  13265.36    გერმანია  January   

   Year  
0  2023  
1  2023  
2  2023  
3  2023  
4  2023

In [11]:
# sorted(df_2023.Partner.unique())

There's extra spaces and wrong country names in df_2023 dataframe's Partner coulmn. for example, `'ავსტრალია ', => 'ავსტრია',` 

In [12]:
# Remove the spaces in the 'Partner' column values.
df_2023['Partner'] = df_2023['Partner'].str.strip()

# Replace wrong country name with the right one.
df_2023['Partner'].replace({'არაბ. გაერთ საამიროები':'არაბთა გაერთიანებული ემირატები',
                            'აშშ':'შეერთებული შტატები',
                            'ბოსნია და გერცოგოვინა':'ბოსნია-ჰერცეგოვინა',
                            'ვირჯინიის კ-ბი(ბრიტ.)':'ვირჯინის კუნძულები, ბრიტანეთი',
                            'კორეის რესპ.':'კორეა',
                            'კოსტა რიკა': 'კოსტარიკა',
                            'კუვეიტი':'ქუვეითი',
                            'ლიხტენშტეინი':'ლიხტენშტაინი',
                            'სამხრეთ აფრიკარ':'სამხრეთ აფრიკა',
                            'საუდების არაბეთი':'საუდის არაბეთი',
                            'შრი-ლანკა':'შრილანკა',
                            'ბელორუსია':'ბელარუსი',
                            "სან-მარინო":'სანმარინო',
                            'სამხრეთი აფრიკა':'სამხრეთ აფრიკა'
                           }, inplace=True)

In [13]:
# Merge df_2023 and countries dataframe to import countries english name column.
df_2023 = pd.merge(df_2023, countries,on='Partner', how='left', sort=False).fillna('Other countries')

# Replace 'Not specified', 'unknown','Europe','EU Countries' with 'Other countries'.
df_2023['Country'].replace(['Not specified', 'unknown','Europe','EU Countries'],'Other countries', inplace=True)

# Replace type column values with English-related words.
df_2023.replace({'Type':{'იმპორტი':'Import', 'ექსპორტი':'Export'}}, inplace=True)

# Drop Partner column
df_2023.drop('Partner', inplace=True, axis=1)

# Create category column.
df_2023['Category'] = df_2023['Code'].astype(str).str[:4]


# Merge df_2023 dataframe to codes dataframe to import categories' names.
df_2023 = pd.merge(df_2023, codes, on='Category', how='left', sort=False).fillna('')

# Show first 5 rows.
df_2023.head()

Code    Type  Quantity  Quantity_kg      Cost    Month  Year  \
0  70051030000  Import  22466.59    222546.67  33782.52  January  2023   
1  39219090000  Import  58782.40     58782.40 547996.80  January  2023   
2  39241000000  Import  10330.41     10330.41  33773.50  January  2023   
3  94016900000  Import     50.00        50.00     64.04  January  2023   
4  83021000009  Import   1150.84      1150.84  13265.36  January  2023   

   Country Category Name  
0     Iran     7005       
1  Türkiye     3921       
2  Türkiye     3924       
3   France     9401       
4  Germany     8302

In [14]:
df_2023.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 405756 entries, 0 to 405755
Data columns (total 10 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Code         405756 non-null  object 
 1   Type         405756 non-null  object 
 2   Quantity     405756 non-null  float64
 3   Quantity_kg  405756 non-null  float64
 4   Cost         405756 non-null  float64
 5   Month        405756 non-null  object 
 6   Year         405756 non-null  object 
 7   Country      405756 non-null  object 
 8   Category     405756 non-null  object 
 9   Name         405756 non-null  object 
dtypes: float64(3), object(7)
memory usage: 31.0+ MB


In [15]:
df_2023.describe()

Quantity  Quantity_kg        Cost
count   405756.00    405756.00   405756.00
mean     23521.09     30121.39    31507.30
std     452395.57   1229843.70   387456.70
min         -2.39        -2.39        0.00
25%          8.10         9.70      268.65
50%         79.40        89.03     1598.11
75%        923.98      1058.35     9730.74
max   54376660.00 315789201.15 67131806.94

# Analysis

### Total Trade Volume

In [16]:
# Calculate Total Export - Import of 2023
total_import_vol = df_2023.query('Type=="Import"')['Cost'].sum()
total_export_vol = df_2023.query('Type=="Export"')['Cost'].sum()

# Calculate balance of trade (BOT)
trade_balance_2023 = total_export_vol - total_import_vol
print(f"{trade_balance_2023:,.1f}")

-7,238,954,818.3


**Georgia has a trade deficit (negative trade balance).**
> "A negative balance of trade means that currency flows outwards to pay for exports, indicating that the country may be overly reliant on foreign goods." - investopedia

### Analyse countries export - import 2022 / January-November 2023

In [17]:
# Add countries share in total export /import in 2022
country_22 = df_2022.groupby(['Country', 'Year'], sort=False).sum(numeric_only=True).fillna(0).reset_index()
country_22['Export_%'] = country_22['Export'].apply(lambda x:x/sum(df_2022['Export'])*100)
country_22['Import_%'] = country_22['Import'].apply(lambda x:x/sum(df_2022['Import'])*100)
country_22.head(5)

Country  Year       Export       Import  Export_%  Import_%
0    Albania  2022    159802.64    513172.35      0.00      0.01
1  Argentina  2022   1137055.43   7955360.93      0.03      0.08
2    Armenia  2022 212639327.81 106555889.38      5.47      1.05
3  Australia  2022   1433696.06   1245664.71      0.04      0.01
4    Austria  2022   2255336.54  63181021.68      0.06      0.63

In [18]:
# Create pivot table of countries export/import in 2023
country_23 = pd.pivot_table(data=df_2023,index=['Country', 'Year'], columns='Type', values='Cost', aggfunc='sum', fill_value=0, sort=False).reset_index()
# Add countries share in total export /import in 2023
country_23['Export_%'] = country_23['Export'].apply(lambda x:x/sum(country_23['Export'])*100).round(3)
country_23['Import_%'] = country_23['Import'].apply(lambda x:x/sum(country_23['Import'])*100).round(3)
country_23.head(5)

Type  Country  Year        Import       Export  Export_%  Import_%
0        Iran  2023  205056927.81  15130176.44      0.55      2.05
1     Türkiye  2023 2022345780.31 293150931.56     10.57     20.20
2      France  2023  210628415.21  43335813.97      1.56      2.10
3     Germany  2023  443164402.83  48520076.91      1.75      4.43
4      Cyprus  2023    5679169.17    704110.71      0.03      0.06

In [19]:
# Merge 2022's and 2023's Dataframes, grouped by countries 
country_23_22 = pd.concat([country_23,country_22])
country_23_22

Country  Year        Import       Export  Export_%  \
0                            Iran  2023  205056927.81  15130176.44      0.55   
1                         Türkiye  2023 2022345780.31 293150931.56     10.57   
2                          France  2023  210628415.21  43335813.97      1.56   
3                         Germany  2023  443164402.83  48520076.91      1.75   
4                          Cyprus  2023    5679169.17    704110.71      0.03   
5                         Armenia  2023   75772505.52 186452949.40      6.72   
6                       Lithuania  2023   57976147.69  22908874.40      0.83   
7                     Netherlands  2023  210382445.53   9191407.94      0.33   
8                        Bulgaria  2023  164185997.41 249008108.93      8.98   
9                  Czech Republic  2023  180945048.80  17664323.99      0.64   
10                          Italy  2023  298700745.37  32568725.77      1.18   
11                         Greece  2023   56533565.88  13776537.51      0.50   
12                        Ukraine  2023  189926664.44  73579054.68      2.65   
13                          China  2023 1141829321.92 268653238.71      9.69   
14                     Uzbekistan  2023   54437509.82  48813219.09      1.76   
15                        Belgium  2023   68984891.81   5087029.42      0.18   
16                     Kazakhstan  2023   25464142.83  66160648.06      2.39   
17                         Poland  2023  184801232.05  31127514.76      1.12   
18                         russia  2023 1651926780.32 514979877.38     18.57   
19                       Portugal  2023   17356154.22   3481368.20      0.13   
20                           Mala  2023   17324778.03     21525.88      0.00   
21                      Hong Kong  2023   90490552.32   5481270.70      0.20   
22                  United States  2023   85613490.02  73925487.67      2.67   
23                    UA Emirates  2023  297814833.20  19175844.49      0.69   
24                       Slovakia  2023   13429940.95   2262780.87      0.08   
25                     Azerbaijan  2023  556088373.72 148212432.21      5.34   
26                        Austria  2023   89048628.09   2168461.82      0.08   
27                          India  2023   95608615.30   9068773.67      0.33   
28                        Morocco  2023   11839935.84  18679215.49      0.67   
29                  Great Britain  2023   98239038.82  17115282.14      0.62   
30                          Spain  2023  171306002.81  74746921.37      2.70   
31                         Sweden  2023   28427747.71   1049926.79      0.04   
32                        Croatia  2023   11135274.29    556219.64      0.02   
33                    Switzerland  2023  118889238.61  79280424.85      2.86   
34                        Hungary  2023   62019201.97   1337705.32      0.05   
35                        Belarus  2023   87507338.82  29290073.66      1.06   
36                         Latvia  2023   36278768.48   8993982.99      0.32   
37                        Ecvador  2023   29482169.44         0.00      0.00   
38                        Estonia  2023    8504756.09   4965161.53      0.18   
39                      Singapore  2023   14277232.83   2036514.28      0.07   
40                  Liechtenstein  2023     568362.93     27000.00      0.00   
41                          Japan  2023   18885168.45   7680654.26      0.28   
42                       Thailand  2023   19440702.96    286781.70      0.01   
43                     Luxembourg  2023    2260051.91     57024.13      0.00   
44                       Slovenia  2023   25399434.35    668895.44      0.02   
45                         Israel  2023   12729003.37  14209338.18      0.51   
46                        Finland  2023   27660793.02    492884.76      0.02   
47                        Ireland  2023   15529098.74    326031.54      0.01   
48                       Paraguay  2023    5681344.96     14697.00      0.00   
49                        

### Main Partner Countries By Export - Import In January - November 2023

In [20]:
# Identify top 10 countries by export/import in 2022 / 2023(excluding December)
top_exporters = country_23_22.groupby('Year', sort=False).apply(lambda x:x.nlargest(10,'Export')).reset_index(drop=True)
top_imorters = country_23_22.groupby('Year', sort=False).apply(lambda x:x.nlargest(10,'Import')).reset_index(drop=True)
top_exporters.head(20)

Country  Year        Import       Export  Export_%  Import_%
0          russia  2023 1651926780.32 514979877.38     18.57     16.50
1         Türkiye  2023 2022345780.31 293150931.56     10.57     20.20
2           China  2023 1141829321.92 268653238.71      9.69     11.40
3        Bulgaria  2023  164185997.41 249008108.93      8.98      1.64
4         Armenia  2023   75772505.52 186452949.40      6.72      0.76
5      Azerbaijan  2023  556088373.72 148212432.21      5.34      5.55
6     Switzerland  2023  118889238.61  79280424.85      2.86      1.19
7           Spain  2023  171306002.81  74746921.37      2.70      1.71
8   United States  2023   85613490.02  73925487.67      2.67      0.85
9         Ukraine  2023  189926664.44  73579054.68      2.65      1.90
10          China  2022 1062562666.38 703502830.03     18.10     10.52
11         russia  2022 1779976789.05 551726786.65     14.19     17.62
12       Bulgaria  2022  190983526.85 423065279.28     10.88      1.89
13        Türkiye  2022 2060897655.54 366507208.60      9.43     20.40
14  United States  2022   91898123.95 263568466.75      6.78      0.91
15        Armenia  2022  106555889.38 212639327.81      5.47      1.05
16     Azerbaijan  2022  620265944.74 166878539.22      4.29      6.14
17           Peru  2022     303266.86 153820385.11      3.96      0.00
18        Ukraine  2022  227858234.82  94889520.31      2.44      2.26
19    Switzerland  2022  143139989.89  90060875.91      2.32      1.42

**In 2023 top coutries by export were:**
- russia with 18.57% in total export,
- Türkiye with 10.57%,
- China with 9.69%.

**In 2022 top coutries by export were:**
- China with 18.10% in total export,
- russia with 14.19%,
- Bulgaria with 10.88%.

In [21]:
# plot data
df = top_exporters.query("Year=='2023'")
# Build the plot
fig = px.scatter(df, x='Country', y='Export', size='Export_%', color='Export_%',
                 color_continuous_scale='mint',
                 size_max=60,
                 labels={'Export': 'Export (USD)', 'Export_%': 'Share in Total Export (%)'},
                 title='MAIN PARTNERS BY EXPORT IN JANUARY - NOVEMBER 2023',
                 text='Country',
                 hover_data={'Export': ':,.2f', 'Export_%': ':.2f%'},
                 
                 )
fig.update_traces(textposition='bottom center')
fig.update_xaxes(showticklabels=False, title='')

fig.show()

In [22]:
df = top_imorters.query("Year=='2023'")
fig = px.scatter(df, x='Country', y='Import', size='Import_%', color='Import_%',
                 color_continuous_scale='orrd',
                 size_max=60,
                 labels={'Import': 'Import (USD)', 'Import_%': 'Share in Total Import (%)'},
                 title='MAIN PARTNERS BY IMPORT IN JANUARY - NOVEMBER 2023',
                 text='Country',
                 hover_data={'Import': ':,.2f', 'Import_%': ':.2f%'},
                 
                 )
fig.update_traces(textposition='bottom center')
fig.update_xaxes(showticklabels=False, title='')

fig.show()

In [23]:
colors = ['#6574F9','#DE413D']
fig = px.bar(top_exporters, x='Country', y='Export',
             text_auto='.4s',
             color='Year',
             color_discrete_sequence = colors,
             labels={'Export_%':'Share in Total (%)', "Export":'Export (USD)'},
             hover_data={'Export': ':,.2f','Export_%':':.2f'},
             title="COMPARE 2022 AND 2023 COUNTRY EXPORT (Excluding December)",
             height=400)
fig.update_layout(
    yaxis=dict(showticklabels=False)
)
# fig.update_layout(yaxis_tickformat='.2f')
fig.show()

In 2023 increased export of russia (4.38%), Türkiye with 1.14%, Armenia(1.25%).

In [24]:
colors = ['#394394','#FCBF41']
fig = px.bar(top_imorters, x='Country', y='Import',
             text_auto='.4s',
             color='Year',
             color_discrete_sequence=colors,
             labels={'Import_%':'Share in Total (%)', "Import":'Import (USD)'},
             hover_data={'Import': ':,.2f','Import_%':':.2f'},
             title="COMPARE 2022 AND 2023 COUNTRY IMPORT (Excluding December)",
             height=400)
fig.update_layout(
    yaxis=dict(showticklabels=False),
)
fig.show()

In 2023 decreased import of russia by 1.54%, increased import of Chine (0.94%) and Germany(0.6%)

## Total Export - Import By Month In 2022 / January-November 2023

In [25]:
month_22 = df_2022.groupby(['Year','Month'], sort=False).sum(numeric_only=True).reset_index()
month_23 = pd.pivot_table(df_2023, index=['Year','Month'],columns='Type', values='Cost' ,aggfunc='sum', sort=False).reset_index()
month_23

Type  Year      Month       Import       Export
0     2023    January 753288670.56 276485448.58
1     2023   February 767696424.12 264247367.15
2     2023      March 996538576.47 290795657.04
3     2023      April 879333542.30 230521680.71
4     2023        May 921919571.00 240554717.84
5     2023       June 971357204.77 257187284.01
6     2023       July 891961799.43 268116353.32
7     2023     August 967491489.98 236236556.22
8     2023  September 944657612.66 254147645.67
9     2023    October 963118902.32 235134279.77
10    2023   November 954252355.24 219234340.20

In [26]:
month_22

Year      Month       Export        Import
0   2022    January 245960982.93  614814418.02
1   2022   February 337198608.38  746805194.42
2   2022      March 346551187.54  699419128.40
3   2022      April 338544337.75  801197294.80
4   2022        May 380635880.70  830029451.44
5   2022       June 282095425.14  839989013.12
6   2022       July 367653344.72  869559676.90
7   2022     August 287980771.00  941637987.55
8   2022  September 349893511.65  882793325.62
9   2022    October 314899982.19  907164242.97
10  2022   November 317551038.18  929799337.22
11  2022   December 318523690.82 1039761570.27

In [27]:
# Metge monthly reports dataframes of 2022 and 2023
month_23_22 = pd.concat([month_23,month_22]).reset_index(drop=True)
#  Calculate export and import shares (in total) and add as columns.
month_23_22['Exp_%'] = month_23_22['Export'].apply(lambda x:x/sum(month_23_22['Export'])*100)
month_23_22['Imp_%'] = month_23_22['Import'].apply(lambda x:x/sum(month_23_22['Import'])*100)
month_23_22

Year      Month        Import       Export  Exp_%  Imp_%
0   2023    January  753288670.56 276485448.58   4.15   3.74
1   2023   February  767696424.12 264247367.15   3.97   3.82
2   2023      March  996538576.47 290795657.04   4.37   4.95
3   2023      April  879333542.30 230521680.71   3.46   4.37
4   2023        May  921919571.00 240554717.84   3.61   4.58
5   2023       June  971357204.77 257187284.01   3.86   4.83
6   2023       July  891961799.43 268116353.32   4.03   4.43
7   2023     August  967491489.98 236236556.22   3.55   4.81
8   2023  September  944657612.66 254147645.67   3.82   4.70
9   2023    October  963118902.32 235134279.77   3.53   4.79
10  2023   November  954252355.24 219234340.20   3.29   4.74
11  2022    January  614814418.02 245960982.93   3.69   3.06
12  2022   February  746805194.42 337198608.38   5.06   3.71
13  2022      March  699419128.40 346551187.54   5.20   3.48
14  2022      April  801197294.80 338544337.75   5.08   3.98
15  2022        May  830029451.44 380635880.70   5.72   4.13
16  2022       June  839989013.12 282095425.14   4.24   4.18
17  2022       July  869559676.90 367653344.72   5.52   4.32
18  2022     August  941637987.55 287980771.00   4.32   4.68
19  2022  September  882793325.62 349893511.65   5.25   4.39
20  2022    October  907164242.97 314899982.19   4.73   4.51
21  2022   November  929799337.22 317551038.18   4.77   4.62
22  2022   December 1039761570.27 318523690.82   4.78   5.17

In [28]:
# plot data
df = month_23
# colors vor bins
colors = ['#394394','#FCBF41']

fig = px.bar(df, x='Month', y=['Export', 'Import'],
             text_auto='.4s',
             color_discrete_map={'Export': colors[0], 'Import': colors[1]},
             labels={'value': 'Cost (USD)', 'variable': 'Trade Type'},
             hover_data={'value': ':,.2f'},
             title='IMPORT - EXPORT IN JANUARY - NOVEMBER 2023',
             height=400
             )

# Format y-axis labels as short numbers
fig.update_layout(
    xaxis=dict(title='Month'),
    yaxis=dict(showticklabels=False),
    )
fig.show()

In [29]:
# Calculate import volume changes between 2022-2023 years
imp_diff = pd.pivot_table(month_23_22, index=['Month'], columns=['Year'], values='Imp_%',sort=False, fill_value=0).reset_index()
imp_diff['YoY'] = imp_diff.apply(lambda x:x['2023']-x['2022'], axis=1)
exp_diff = pd.pivot_table(month_23_22, index=['Month'], columns=['Year'], values='Exp_%',sort=False,fill_value=0).reset_index()
exp_diff['YoY'] = exp_diff.apply(lambda x:x['2023']-x['2022'], axis=1)
exp_diff

Year      Month  2023  2022   YoY
0       January  4.15  3.69  0.46
1      February  3.97  5.06 -1.10
2         March  4.37  5.20 -0.84
3         April  3.46  5.08 -1.62
4           May  3.61  5.72 -2.10
5          June  3.86  4.24 -0.37
6          July  4.03  5.52 -1.49
7        August  3.55  4.32 -0.78
8     September  3.82  5.25 -1.44
9       October  3.53  4.73 -1.20
10     November  3.29  4.77 -1.48
11     December  0.00  4.78 -4.78

In [30]:
fig = go.Figure()
m1 = month_23_22.query('Year=="2023"')
m2 = month_23_22.query('Year=="2022"')
fig.add_trace(
    go.Scatter(
        x=imp_diff['Month'],
        y=imp_diff['YoY'],
        mode='lines+markers',
        text=[f"{y:.2f}%" for y in exp_diff['YoY']], 
        name='YoY'
    ))
fig.add_trace(go.Bar(
    x=m1['Month'],
    y=m1['Imp_%'],
    name='2023',
    marker_color='#FF6692'
))
fig.add_trace(go.Bar(
    x=m2['Month'],
    y=m2['Imp_%'],
    name='2022',
    marker_color='#B6E880'
))

fig.update_layout(
    barmode='group',
    title='MONTHLY IMPORT COMPARISION (2022 vs 2023)',
    yaxis=dict(
        tickformat='.2f',
        title='Import (%)'
    )
)
fig.show()

In [31]:
fig = go.Figure()
fig.add_trace(
    go.Scatter(
        x=exp_diff['Month'],
        y=exp_diff['YoY'],
        mode='lines+markers',
        text=[f"{y:.2f}%" for y in exp_diff['YoY']],  
        name='YoY'
    ))
fig.add_trace(go.Bar(
    x=m1['Month'],
    y=m1['Exp_%'],
    name='2023',
    marker_color='#FF6692'
))
fig.add_trace(go.Bar(
    x=m2['Month'],
    y=m2['Exp_%'],
    name='2022',
    marker_color='#B6E880'
))
fig.update_layout(
    barmode='group',
    title='MONTHLY EXPORT COMPARISION (2022 vs. 2023)',
    yaxis=dict(
        tickformat='.2f',
        title='Export'
    )

)
fig.show()

In [32]:
fig = go.Figure()

fig.add_trace(go.Scatterpolar(
      r=month_23['Export'],
      theta=month_23['Month'],
      fill='toself',
    hovertemplate='%{r:,.2f}',
    mode='markers',
      name='2023'
))
fig.add_trace(go.Scatterpolar(
      r=month_22['Export'],
      theta=month_22['Month'],
      fill='tonext',
    hovertemplate='%{r:,.2f}',
    mode='markers',
      name='2022'
))

fig.update_layout(
    title='COMPARISION OF MONTHLY EXPORTS IN 2022 AND 2023',
  polar=dict(
    radialaxis=dict(
      visible=True,
    tickformat='.0s',
      range=[month_22['Export'], month_23['Export']]
    )),
)

fig.show()

In [33]:
fig = go.Figure()

fig.add_trace(go.Scatterpolar(
      r=month_23['Import'],
      theta=month_23['Month'],
      fill='toself',
    hovertemplate='%{r:,.2f}',
    mode='markers',
      name='2023'
))
fig.add_trace(go.Scatterpolar(
      r=month_22['Import'],
      theta=month_22['Month'],
    hovertemplate='%{r:,.2f}',
    mode='markers',
      fill='tonext',  
      name='2022'
))

fig.update_layout(
    title="COMPARISION OF MONTHLY IMPORTS IN 2022 AND 2023",
    polar=dict(
    radialaxis=dict(
      visible=True,
        tickformat='.0s',
      range=[month_22['Import'], month_23['Import']]
    )),
)

fig.show()

#### Import increased in 2023, but decreased export (exception Jan, 2022)

##  The major product categories being exported and imported.

#### Main Imported Product Categories in January - November 2023

In [34]:
# group df_2023 dataframe by category code
grouped_cat = pd.pivot_table(df_2023,index='Category', columns = 'Type', values='Cost', aggfunc='sum', fill_value=0).reset_index()
grouped_cat.head(10)

Type Category      Export      Import
0        0101     3000.00    45262.83
1        0102 73733139.39 22647994.52
2        0103    24962.20 24848228.58
3        0104 42904875.99  1597456.53
4        0105  6964221.50  6578667.78
5        0106    50101.77   434612.41
6        0201     5848.05  5793033.66
7        0202   111706.23 15817016.52
8        0203     5145.41 50218995.20
9        0204  1901569.99    70680.50

In [35]:
# main product categories by import / export in 2023 (Jan-Nov)
top_imp_cat = grouped_cat.sort_values('Import', ascending=False).fillna(0)
top_exp_cat = grouped_cat.sort_values('Export', ascending=False).fillna(0)
top_imp_cat = pd.merge(top_imp_cat,codes, on='Category', how='outer').fillna('')
top_exp_cat = pd.merge(top_exp_cat,codes, on='Category', how='outer').fillna('')

# Let's look at top exported product categories datafreame
top_exp_cat.head()

Category       Export      Import  \
0     2603 487347726.51        0.00   
1     2204 239916256.74  4105681.64   
2     7202 150088987.50  2269415.92   
3     2208 149509610.36 28590003.20   
4     3102 147103921.34  9873894.11   

                                           Name  
0                  Copper ores and concentrates  
1                                          Wine  
2                                  Ferro-alloys  
3                     Undenatured ethyl alcohol  
4  Mineral or chemical fertilisers, nitrogenous

In [36]:
# Top imported product categories datafreame
top_imp_cat.head()

Category      Export        Import  \
0     2710 26766021.35 1024881705.33   
1     2711   183540.29  448448201.88   
2     3004 51036626.89  375043010.37   
3     8517  5521624.75  327556812.63   
4     8471  6416496.08  134224969.23   

                                             Name  
0                                  Petroleum oils  
1  Petroleum gases and other gaseous hydrocarbons  
2                                     Medicaments  
3                       Telephone sets and others  
4              Automatic data-processing machines

In [37]:
top_imp_cat.loc[top_imp_cat['Import'] < 70564528.53, 'Name'] = 'Other categories' # Represent only large countries
fig = px.pie(top_imp_cat, values='Import', names='Name', title='MAIN PRODUCT CATEGORIES BY IMPORT JANUARY - NOVEMBER 2023')
fig.show()

Top imported product categories:
- Petrolium oil(10.2%)
- Petrolium gases (4.48%)
- Medicaments (3.75%)

In [38]:
top_exp_cat.loc[top_exp_cat['Export'] < 40000000, 'Name'] = 'Other categories'
fig = px.pie(top_exp_cat, values='Export', names='Name', title='MAIN PRODUCT CATEGORIES BY EXPORT JANUARY - NOVEMBER 2023')
fig.show()

Top exported product categories:
- Copper ores and concentrates (17.6%)
- Wine (8.65%)
- Ferro-alloys (5.41%)

### Let's determine the share of each country in the top imported categories.

In [39]:
# filter df_2023 dataframe by import and group by code and country
pt = df_2023[df_2023['Type']=='Import'].groupby(['Category', 'Country'])['Cost'].sum(numeric_only=True).unstack().reset_index().fillna(0).dropna(how='all',axis=1)
# calculate total of each record and append to dataframe
pt['Total'] = pt.sum(numeric_only=True,axis=1)

In [40]:
pt.head()

Country Category  Albania  Algeria  Andorra  Angola  Argentina   Armenia  \
0           0101     0.00     0.00     0.00    0.00       0.00      0.00   
1           0102     0.00     0.00     0.00    0.00       0.00      0.00   
2           0103     0.00     0.00     0.00    0.00       0.00      0.00   
3           0104     0.00     0.00     0.00    0.00       0.00 165362.17   
4           0105     0.00     0.00     0.00    0.00       0.00      0.00   

Country  Australia   Austria  Azerbaijan  Bahrain  Bangladesh   Belarus  \
0             0.00      0.00     4600.00     0.00        0.00      0.00   
1             0.00 181152.79        0.00     0.00        0.00      0.00   
2             0.00      0.00        0.00     0.00        0.00 150804.90   
3             0.00      0.00        0.00     0.00        0.00      0.00   
4             0.00      0.00        0.00     0.00        0.00      0.00   

Country   Belgium  Benin  Bosnia-Herzegovina  Brasil  Bulgaria  Cambodia  \
0            0.00   0.00                0.00    0.00   7832.28      0.00   
1            0.00   0.00                0.00    0.00      0.00      0.00   
2            0.00   0.00                0.00    0.00      0.00      0.00   
3            0.00   0.00                0.00    0.00      0.00      0.00   
4       242305.47   0.00                0.00    0.00      0.00      0.00   

Country  Cameroon  Canada  Chile  China  Colombia  Congo  Costa Rica  Croatia  \
0            0.00    0.00   0.00   0.00      0.00   0.00        0.00     0.00   
1            0.00    0.00   0.00   0.00      0.00   0.00        0.00     0.00   
2            0.00    0.00   0.00   0.00      0.00   0.00        0.00     0.00   
3            0.00    0.00   0.00   0.00      0.00   0.00        0.00     0.00   
4            0.00    0.00   0.00   0.00      0.00   0.00        0.00     0.00   

Country  Cyprus  Czech Republic    Denmark  Domineika  Ecvador  Egypt  \
0          0.00            0.00       0.00       0.00     0.00   0.00   
1          0.00       548314.47 5408719.79       0.00     0.00   0.00   
2          0.00            0.00       0.00       0.00     0.00   0.00   
3          0.00            0.00       0.00       0.00     0.00   0.00   
4          0.00         5878.73       0.00       0.00     0.00   0.00   

Country    Estonia  Ethiopia  Finland    France  Gabon  Gambia  Gana  Georgia  \
0             0.00      0.00     0.00      0.00   0.00    0.00  0.00     0.00   
1       1518458.53      0.00     0.00 488513.66   0.00    0.00  0.00     0.00   
2             0.00      0.00     0.00      0.00   0.00    0.00  0.00     0.00   
3             0.00      0.00     0.00      0.00   0.00    0.00  0.00     0.00   
4             0.00      0.00     0.00  49860.23   0.00    0.00  0.00     0.00   

Country  Germany  Gibraltar  Great Britain  Greece  Guatemala  Hong Kong  \
0           0.00       0.00           0.00    0.00       0.00       0.00   
1           0.00       0.00           0.00    0.00       0.00       0.00   
2           0.00       0.00           0.00    0.00       0.00       0.00   
3           0.00       0.00      135052.93    0.00       0.00       0.00   
4       20402.30       0.00           0.00    0.00       0.00       0.00   

Country    Hungary  Iceland  India  Indonesia  Iran  Iraq  Ireland  Israel  \
0             0.00     0.00   0.00       0.00  0.00  0.00     0.00    0.00   
1        462799.34     0.00   0.00       0.00  0.00  0.00     0.00    0.00   
2             0.00     0.00   0.00       0.00  0.00  0.00     0.00    0.00   
3             0.00     0.00   0.00       0.00  0.00  0.00     0.00    0.00   
4       1559356.70     0.00   0.00       0.00  0.00  0.00     0.00    0.00   

Country  Italy  Jamaica  Japan  Jordan  Kazakhstan  Kenya  Kuwait  Kyrgyzstan  \
0         0.00     0.00   0.00    0.00        0.00   0.00    0.00        0.00   
1         0.00     0.00   0.00    0.00        0.00   0.00    0.00        0.00   
2         0.00     0.00   0.00    0.00        0.00   

In [41]:
pt=pt.sort_values('Total', ascending=False)[:10]

In [42]:
# Remove empty-valued column
country_share_imp = pt.loc[:,(pt != 0).any(axis=0)].reset_index(-1, drop=True)
# Calculate percentage
country_share_imp.iloc[:,1:-1] = country_share_imp.iloc[:,1:-1].apply(lambda x: x.div(x.sum()).mul(100), axis=1)
country_share_imp

Country Category  Albania  Argentina  Armenia  Australia  Austria  Azerbaijan  \
0           2710     0.03       0.00     0.00       0.00     0.00        7.84   
1           2711     0.00       0.00     0.00       0.00     0.00       63.91   
2           3004     0.00       0.55     0.59       0.00     0.72        0.00   
3           8517     0.00       0.00     0.00       0.00     0.24        0.01   
4           8471     0.00       0.00     0.33       0.00     0.04        0.07   
5           7214     0.00       0.00     0.00       0.00     0.00       25.60   
6           7308     0.00       0.00     0.12       0.00     6.70        1.42   
7           8429     0.00       0.00     0.01       0.02     0.95        0.70   
8           1001     0.00       0.00     0.00       0.00     0.17        0.00   
9           2402     0.00       0.00     8.86       0.00     0.00        2.79   

Country  Bangladesh  Belarus  Belgium  Bosnia-Herzegovina  Bulgaria  Cambodia  \
0              0.00     2.15     0.36                0.00      7.26      0.00   
1              0.00     0.01     0.00                0.00      0.00      0.00   
2              0.25     1.45     0.62                0.08      1.98      0.00   
3              0.00     0.02     0.01                0.00      0.04      0.00   
4              0.00     0.12     0.19                0.00      0.19      0.00   
5              0.00     0.00     0.00                0.00      0.00      0.00   
6              0.00     0.96     0.00                0.00      0.12      0.00   
7              0.00     0.01     1.12                0.00      7.61      1.32   
8              0.00     0.00     0.00                0.00      0.00      0.00   
9              0.00     0.00     0.07                0.00      1.30      0.00   

Country  Canada  Chile  China  Croatia  Cyprus  Czech Republic  Denmark  \
0          0.00   0.00   0.01     0.00    0.00            0.01     0.00   
1          0.00   0.00   0.00     0.00    0.00            0.00     0.00   
2          0.12   0.03   0.08     0.19    0.10            0.44     0.34   
3          0.00   0.00   3.15     0.00    0.73           21.38     0.01   
4          0.00   0.00  33.22     0.00    0.44           14.46     0.02   
5          0.00   0.00   0.00     0.00    0.00            0.00     0.00   
6          0.00   0.00   8.65     0.00    0.00            0.27     0.00   
7          0.44   0.00  18.33     0.00    0.00            1.03     0.00   
8          0.00   0.00   0.00     0.00    0.00            0.00     0.00   
9          0.00   0.00   0.00     0.00    0.00            5.58     0.00   

Country  Domineika  Egypt  Estonia  Finland  France  Georgia  Germany  \
0             0.00   0.00     0.00     0.15    0.14     0.03     1.20   
1             0.00   0.00     0.00     0.00    0.00     0.00     0.00   
2             0.00   0.05     0.01     0.12    7.43     0.48    13.65   
3             0.00   0.00     0.00     0.67    0.04     0.00     0.75   
4             0.01   0.00     0.09     0.26    0.14     0.00     7.91   
5             0.00   0.00     0.00     0.00    0.00     0.00     0.01   
6             0.00   0.00     0.00     0.01    0.11     0.00     1.64   
7             0.00   0.00     0.00     0.00    0.06     0.00     2.35   
8             0.00   0.00     0.00     0.00    0.03     0.00     0.12   
9             0.05   0.00     0.00     0.00    0.00     0.00     0.32   

Country  Great Britain  Greece  Hong Kong  Hungary  India  Indonesia  Iran  \
0                 0.11    3.37       0.00     0.08   0.01       0.00  0.16   
1                 0.00    0.00       0.01     0.00   0.00       0.00  0.03   
2                 1.08    0.30       0.00     9.73   2.98       0.00  0.01   
3                 0.14    0.00       6.17     0.00   1.77       0.00  0.00   
4                 0.42    0.02      12.39     0.11   0.00       0.00  0.00   
5                 0.00    0.00       0.00     0.00   0.00       0.00 28.31   
6                 0.12    0.02       0.3

In [43]:
pd.merge(codes, country_share_imp, on='Category').sort_values('Total', ascending=False).head()

Category                                            Name  Albania  \
7     2710                                  Petroleum oils     0.03   
0     2711  Petroleum gases and other gaseous hydrocarbons     0.00   
1     3004                                     Medicaments     0.00   
2     8517                       Telephone sets and others     0.00   
3     8471              Automatic data-processing machines     0.00   

   Argentina  Armenia  Australia  Austria  Azerbaijan  Bangladesh  Belarus  \
7       0.00     0.00       0.00     0.00        7.84        0.00     2.15   
0       0.00     0.00       0.00     0.00       63.91        0.00     0.01   
1       0.55     0.59       0.00     0.72        0.00        0.25     1.45   
2       0.00     0.00       0.00     0.24        0.01        0.00     0.02   
3       0.00     0.33       0.00     0.04        0.07        0.00     0.12   

   Belgium  Bosnia-Herzegovina  Bulgaria  Cambodia  Canada  Chile  China  \
7     0.36                0.00      7.26      0.00    0.00   0.00   0.01   
0     0.00                0.00      0.00      0.00    0.00   0.00   0.00   
1     0.62                0.08      1.98      0.00    0.12   0.03   0.08   
2     0.01                0.00      0.04      0.00    0.00   0.00   3.15   
3     0.19                0.00      0.19      0.00    0.00   0.00  33.22   

   Croatia  Cyprus  Czech Republic  Denmark  Domineika  Egypt  Estonia  \
7     0.00    0.00            0.01     0.00       0.00   0.00     0.00   
0     0.00    0.00            0.00     0.00       0.00   0.00     0.00   
1     0.19    0.10            0.44     0.34       0.00   0.05     0.01   
2     0.00    0.73           21.38     0.01       0.00   0.00     0.00   
3     0.00    0.44           14.46     0.02       0.01   0.00     0.09   

   Finland  France  Georgia  Germany  Great Britain  Greece  Hong Kong  \
7     0.15    0.14     0.03     1.20           0.11    3.37       0.00   
0     0.00    0.00     0.00     0.00           0.00    0.00       0.01   
1     0.12    7.43     0.48    13.65           1.08    0.30       0.00   
2     0.67    0.04     0.00     0.75           0.14    0.00       6.17   
3     0.26    0.14     0.00     7.91           0.42    0.02      12.39   

   Hungary  India  Indonesia  Iran  Iraq  Ireland  Israel  Italy  Japan  \
7     0.08   0.01       0.00  0.16  0.00     0.00    0.00   0.12   0.03   
0     0.00   0.00       0.00  0.03  0.00     0.00    0.00   0.00   0.00   
1     9.73   2.98       0.00  0.01  0.00     0.31    0.02   2.27   0.02   
2     0.00   1.77       0.00  0.00  0.00     0.02    0.00   0.01   0.00   
3     0.11   0.00       0.00  0.00  0.00     1.02    0.09   0.26   0.01   

   Kazakhstan  Kyrgyzstan  Latvia  Lebanon  Libia  Liechtenstein  Lithuania  \
7        0.01        0.00    0.00     0.00   0.42           0.00       0.19   
0        0.00        0.00    0.00     0.00   0.00           0.00       0.00   
1        0.00        0.00    1.40     0.01   0.00           0.00       3.33   
2        0.08        0.00    0.16     0.00   0.00           0.00       0.21   
3        0.00        0.00    0.31     0.00   0.00           0.00       0.30   

   Luxembourg  Macedonia  Mala  Malaysia  Mali  Moldova  Morocco  Nepal  \
7        0.00       0.00  0.80      0.00  0.00     0.00     0.00   0.00   
0        0.00       0.00  0.00      0.00  0.00     0.00     0.00   0.00   
1        0.00       0.02  0.10      0.00  0.00     0.43     0.01   0.00   
2        0.00       0.00  0.00      0.02  0.00     0.00     0.00   0.00   
3        0.01       0.00  0.00      0.01  0.00     0.00     0.00   0.00   

   Netherlands  Nicaragua  Nigeria  Norway  Oman  Other countries  Pacistan  \
7         0.28       0.00     0.00    0.00  0.00             0.01      0.00   
0         0.00       0.00     0.00    0.00  0.00             0.00      0.00   
1         2.65       0.00     0.00    0.00  0.00             0.02      0.15   
2         2.98       0.00     0.00    0.00  0.00             0.00 

In [44]:
# Take a glance at the first record (the 1st most imported product category)
country_share_imp.iloc[:1,:].replace({0:np.nan}).dropna(axis=1)

Country Category  Albania  Armenia  Austria  Azerbaijan  Belarus  Belgium  \
0           2710     0.03     0.00     0.00        7.84     2.15     0.36   

Country  Bulgaria  Canada  China  Czech Republic  Denmark  Estonia  Finland  \
0            7.26    0.00   0.01            0.01     0.00     0.00     0.15   

Country  France  Georgia  Germany  Great Britain  Greece  Hong Kong  Hungary  \
0          0.14     0.03     1.20           0.11    3.37       0.00     0.08   

Country  India  Iran  Iraq  Israel  Italy  Japan  Kazakhstan  Latvia  Libia  \
0         0.01  0.16  0.00    0.00   0.12   0.03        0.01    0.00   0.42   

Country  Lithuania  Luxembourg  Mala  Mali  Nepal  Netherlands  \
0             0.19        0.00  0.80  0.00   0.00         0.28   

Country  Other countries  Poland  Romania  Saudi Arabia  Serbia  Singapore  \
0                   0.01    0.11    13.82          0.00    0.05       0.03   

Country  Slovakia  Slovenia  South Korea  Spain  Sweden  Switzerland  \
0            0.00      0.00         0.05   0.02    0.02         0.19   

Country  Thailand  The U.S. Virgin Islands  Turkmenistan  Türkiye  \
0            0.05                     0.18          7.15     3.30   

Country  UA Emirates  Ukraine  United States  russia         Total  
0               0.27     0.08           0.07   49.83 1024881705.33

This table illustrates the countries' shares of the top imported products in total. For instance, currently, **49.83%** of the most imported product, Petroleum oil *(code: "p2710")*, comes from russia. The second position is held by Romania, contributing **13.82%** to the import of petroleum oil. Following closely is Azerbaijan, accounting for **7.84%**.

In [45]:
# The second most imported product.
country_share_imp.iloc[1:2,:].replace({0:np.nan}).dropna(axis=1)

Country Category  Azerbaijan  Belarus  China  France  Germany  Great Britain  \
1           2711       63.91     0.01   0.00    0.00     0.00           0.00   

Country  Greece  Hong Kong  Iran  Ireland  Italy  Liechtenstein  Netherlands  \
1          0.00       0.01  0.03     0.00   0.00           0.00         0.00   

Country  Poland  Singapore  South Korea  Spain  Türkiye  UA Emirates  russia  \
1          0.00       0.00         0.02   0.00     0.00         0.00   36.01   

Country        Total  
1       448448201.88

Of the most imported product, Petroleum oil *(code: 'p2711')*, **63.91%** comes from Azerbaijan, and the remaining **36%** is imported from russia.

### The share of each country in the top exported categories.

Apply the same modifications to the DataFrame df_2023 as previously done for imported categories above to ascertain each country's share of top exported categories, now focusing on export data.

In [46]:
ptx = df_2023.query('Type=="Export"').groupby(['Category', 'Country'])['Cost'].sum().unstack().reset_index().fillna(0).dropna(how='all',axis=1)
ptx['Total'] = ptx.sum(numeric_only=True, axis=1)
ptx.head()

Country Category  Albania  Algeria  Argentina    Armenia  Australia  Austria  \
0           0101     0.00     0.00       0.00    3000.00       0.00     0.00   
1           0102     0.00     0.00       0.00  348442.90       0.00     0.00   
2           0103     0.00     0.00       0.00   24962.20       0.00     0.00   
3           0104     0.00     0.00       0.00       0.00       0.00     0.00   
4           0105     0.00     0.00       0.00 2334639.50       0.00     0.00   

Country  Azerbaijan  Bahrain  Belarus  Belgium  Belize  Bhutan  \
0              0.00     0.00     0.00     0.00    0.00    0.00   
1       18084905.40     0.00     0.00     0.00    0.00    0.00   
2              0.00     0.00     0.00     0.00    0.00    0.00   
3       14482807.40     0.00     0.00     0.00    0.00    0.00   
4        4214294.50     0.00     0.00     0.00    0.00    0.00   

Country  Bosnia-Herzegovina  Botswana  Brasil  Bulgaria  Burkinafaso  Burundi  \
0                      0.00      0.00    0.00      0.00         0.00     0.00   
1                      0.00      0.00    0.00      0.00         0.00     0.00   
2                      0.00      0.00    0.00      0.00         0.00     0.00   
3                      0.00      0.00    0.00      0.00         0.00     0.00   
4                      0.00      0.00    0.00      0.00         0.00     0.00   

Country  Cambodia  Cameroon  Canada  Chad  Chile  China  Colombia  Congo  \
0            0.00      0.00    0.00  0.00   0.00   0.00      0.00   0.00   
1            0.00      0.00    0.00  0.00   0.00   0.00      0.00   0.00   
2            0.00      0.00    0.00  0.00   0.00   0.00      0.00   0.00   
3            0.00      0.00    0.00  0.00   0.00   0.00      0.00   0.00   
4            0.00      0.00    0.00  0.00   0.00   0.00      0.00   0.00   

Country  Costa Rica  Croatia  Cuba  Cyprus  Czech Republic  Côte d'Ivoire  \
0              0.00     0.00  0.00    0.00            0.00           0.00   
1              0.00     0.00  0.00    0.00            0.00           0.00   
2              0.00     0.00  0.00    0.00            0.00           0.00   
3              0.00     0.00  0.00    0.00            0.00           0.00   
4              0.00     0.00  0.00    0.00            0.00           0.00   

Country  Denmark  Egypt  Estonia  Ethiopia  Finland  France  Gana  Georgia  \
0           0.00   0.00     0.00      0.00     0.00    0.00  0.00     0.00   
1           0.00   0.00     0.00      0.00     0.00    0.00  0.00     0.00   
2           0.00   0.00     0.00      0.00     0.00    0.00  0.00     0.00   
3           0.00   0.00     0.00      0.00     0.00    0.00  0.00     0.00   
4           0.00   0.00     0.00      0.00     0.00    0.00  0.00     0.00   

Country  Germany  Great Britain  Greece  Guinea  Honduras  Hong Kong  Hungary  \
0           0.00           0.00    0.00    0.00      0.00       0.00     0.00   
1           0.00           0.00    0.00    0.00      0.00       0.00     0.00   
2           0.00           0.00    0.00    0.00      0.00       0.00     0.00   
3           0.00           0.00    0.00    0.00      0.00       0.00     0.00   
4           0.00           0.00    0.00    0.00      0.00       0.00     0.00   

Country  Iceland  India  Indonesia       Iran        Iraq  Ireland  Israel  \
0           0.00   0.00       0.00       0.00        0.00     0.00    0.00   
1           0.00   0.00       0.00  377172.18 54922618.91     0.00    0.00   
2           0.00   0.00       0.00       0.00        0.00     0.00    0.00   
3           0.00   0.00       0.00 1050252.28  7415392.31     0.00    0.00   
4           0.00   0.00       0.00  415287.50        0.00     0.00    0.00   

Country  Italy  Japan  Jordan  Kazakhstan  Kenya     Kuwait  Kyrgyzstan  Laos  \
0         0.00   0.00    0.00        0.00   0.00       0.00        0.00  0.00   
1         0.00   0.00    0.00        0.00   0.00       0.00        0.00  0.00   
2         0.00   0.00    0.00        0.00   0.0

In [47]:
ptx = ptx.sort_values('Total', ascending=False).reset_index(-1, drop=True)[:10]
ptx

Country Category  Albania    Algeria  Argentina     Armenia  Australia  \
0           2603     0.00       0.00       0.00        0.00       0.00   
1           2204     0.00       0.00       0.00    67210.48  104950.14   
2           7202     0.00 5827500.00       0.00   245215.63       0.00   
3           2208     0.00       0.00       0.00 18344491.65    1581.12   
4           3102     0.00       0.00       0.00  1259557.03    1620.51   
5           2201     0.00       0.00       0.00   193626.48   21799.80   
6           2202     0.00       0.00       0.00 10703400.18   12506.04   
7           7108     0.00       0.00       0.00        0.00       0.00   
8           0802     0.00  362747.92  628350.00  4239726.04   52450.00   
9           0102     0.00       0.00       0.00   348442.90       0.00   

Country   Austria  Azerbaijan  Bahrain     Belarus   Belgium  Belize  Bhutan  \
0            0.00        0.00     0.00        0.00      0.00    0.00    0.00   
1       160651.97   578532.48     0.00  7584706.74 304144.99    0.00    0.00   
2       217440.00  6541585.81     0.00        0.00      0.00    0.00    0.00   
3        14023.48    51147.16     0.00 16064354.84    210.12    0.00    0.00   
4            0.00   915314.10     0.00        0.00      0.00    0.00    0.00   
5            4.97  1394621.31     0.00  1786296.84    648.00    0.00    0.00   
6        52352.40 12287588.76     0.00   490339.23      0.00    0.00    0.00   
7            0.00        0.00     0.00        0.00      0.00    0.00    0.00   
8        49405.01   541145.50     0.00   782762.54 444100.00    0.00    0.00   
9            0.00 18084905.40     0.00        0.00      0.00    0.00    0.00   

Country  Bosnia-Herzegovina  Botswana   Brasil     Bulgaria  Burkinafaso  \
0                      0.00      0.00     0.00 239967738.80         0.00   
1                      0.00      0.00 19217.69    183897.05         0.00   
2                      0.00      0.00     0.00         0.00         0.00   
3                      0.00      0.00     0.00     60974.93         0.00   
4                      0.00      0.00     0.00   1755600.00         0.00   
5                      0.00      0.00     0.00    207356.96         0.00   
6                      0.00      0.00     0.00    236554.42     49524.26   
7                      0.00      0.00     0.00         0.00         0.00   
8                 306561.03      0.00     0.00     48345.28         0.00   
9                      0.00      0.00     0.00         0.00         0.00   

Country  Burundi  Cambodia  Cameroon    Canada  Chad  Chile        China  \
0           0.00      0.00      0.00     86.42  0.00   0.00 190888955.94   
1           0.00      0.00      0.00 790567.28  0.00   0.00  10306004.69   
2           0.00      0.00      0.00      0.00  0.00   0.00         0.00   
3           0.00      0.00      0.00  29881.80  0.00   0.00    527711.77   
4           0.00      0.00      0.00      0.00  0.00   6.43         0.00   
5           0.00      0.00      0.00 376633.02  0.00   0.00    264931.43   
6           0.00      0.00      0.00  35178.88  0.00   0.00    112289.21   
7           0.00      0.00      0.00      0.00  0.00   0.00         0.00   
8           0.00      0.00      0.00 449062.51  0.00   0.00        64.56   
9           0.00      0.00      0.00      0.00  0.00   0.00         0.00   

Country  Colombia    Congo  Costa Rica  Croatia  Cuba    Cyprus  \
0            0.00     0.00        0.00     0.00  0.00      0.00   
1            0.00 26423.08        0.00     0.00  0.00  68061.42   
2            0.00     0.00        0.00     0.00  0.00 122400.00   
3            0.00  9908.72        0.00     0.00  0.00   4350.25   
4            0.00     0.00        0.00     0.00  0.00      0.00   
5            0.00     0.00        0.00     0.00  0.00 216793.02   
6            0.00 97779.57        0.00     0.00  0.00  73484.26   
7            0.00     0.00        0.00     0.00  0.00      0.00   
8            0.00     0.00 

In [48]:
country_share_exp = ptx.loc[:,(ptx != 0).any(axis=0)].reset_index(-1, drop=True)
country_share_exp.iloc[:,1:-1] = country_share_exp.iloc[:,1:-1].apply(lambda x: x.div(x.sum()).mul(100), axis=1)
country_share_exp

Country Category  Algeria  Argentina  Armenia  Australia  Austria  Azerbaijan  \
0           2603     0.00       0.00     0.00       0.00     0.00        0.00   
1           2204     0.00       0.00     0.03       0.04     0.07        0.24   
2           7202     3.88       0.00     0.16       0.00     0.14        4.36   
3           2208     0.00       0.00    12.27       0.00     0.01        0.03   
4           3102     0.00       0.00     0.86       0.00     0.00        0.62   
5           2201     0.00       0.00     0.16       0.02     0.00        1.15   
6           2202     0.00       0.00    10.38       0.01     0.05       11.92   
7           7108     0.00       0.00     0.00       0.00     0.00        0.00   
8           0802     0.48       0.83     5.59       0.07     0.07        0.71   
9           0102     0.00       0.00     0.47       0.00     0.00       24.53   

Country  Belarus  Belgium  Bosnia-Herzegovina  Brasil  Bulgaria  Burkinafaso  \
0           0.00     0.00                0.00    0.00     49.24         0.00   
1           3.16     0.13                0.00    0.01      0.08         0.00   
2           0.00     0.00                0.00    0.00      0.00         0.00   
3          10.74     0.00                0.00    0.00      0.04         0.00   
4           0.00     0.00                0.00    0.00      1.19         0.00   
5           1.47     0.00                0.00    0.00      0.17         0.00   
6           0.48     0.00                0.00    0.00      0.23         0.05   
7           0.00     0.00                0.00    0.00      0.00         0.00   
8           1.03     0.59                0.40    0.00      0.06         0.00   
9           0.00     0.00                0.00    0.00      0.00         0.00   

Country  Canada  Chile  China  Congo  Cyprus  Czech Republic  Côte d'Ivoire  \
0          0.00   0.00  39.17   0.00    0.00            0.00           0.00   
1          0.33   0.00   4.30   0.01    0.03            0.11           0.00   
2          0.00   0.00   0.00   0.00    0.08            0.00           0.00   
3          0.02   0.00   0.35   0.01    0.00            0.11           0.00   
4          0.00   0.00   0.00   0.00    0.00            0.00           0.00   
5          0.31   0.00   0.22   0.00    0.18            0.03           0.00   
6          0.03   0.00   0.11   0.09    0.07            0.02           0.07   
7          0.00   0.00   0.00   0.00    0.00            0.00           0.00   
8          0.59   0.00   0.00   0.00    0.00            3.63           0.00   
9          0.00   0.00   0.00   0.00    0.00            0.00           0.00   

Country  Denmark  Egypt  Estonia  Finland  France  Gana  Georgia  Germany  \
0           0.00   0.00     0.00     0.00    0.00  0.00     0.00     0.00   
1           0.03   0.00     0.66     0.04    0.22  0.00     0.00     1.84   
2           0.00   1.53     0.00     0.00    1.59  0.00     0.00     0.09   
3           0.00   0.00     0.13     0.00   11.18  0.00     0.87     0.09   
4           0.00   0.00     0.00     0.00    0.00  4.54     0.00     0.00   
5           0.00   0.00     0.00     0.00    0.00  0.00     0.02     1.57   
6           0.00   0.00     0.22     0.00    0.05  0.00     0.00     0.53   
7           0.00   0.00     0.00     0.00    0.00  0.00     0.00     0.00   
8           0.00   0.00     0.23     0.00    5.03  0.00     0.00    13.58   
9           0.00   0.00     0.00     0.00    0.00  0.00     0.00     0.00   

Country  Great Britain  Greece  Guinea  Hong Kong  Hungary  Iceland  India  \
0                 0.00    0.00    0.00       0.00     0.00     0.00   0.00   
1                 0.53    0.01    0.00       0.04     0.01     0.00   0.00   
2                 0.12    0.00    0.00       0.00     0.00     0.00   0.00   
3                 0.02    0.00    0.00       0.01     0.00     0.00   0.00   
4                 0.00    0.00    0.00       0.00     0.00     0.00   4.69   
5                 0.26    0.11    0.00   

In [49]:
# Merge with 'codes' DataFrame to add a 'name' column based on a common code.
pd.merge(codes,country_share_exp,on='Category').sort_values('Total', ascending=False).head()

Category                                          Name  Algeria  Argentina  \
0     2603                  Copper ores and concentrates     0.00       0.00   
5     2204                                          Wine     0.00       0.00   
2     7202                                  Ferro-alloys     3.88       0.00   
7     2208                     Undenatured ethyl alcohol     0.00       0.00   
1     3102  Mineral or chemical fertilisers, nitrogenous     0.00       0.00   

   Armenia  Australia  Austria  Azerbaijan  Belarus  Belgium  \
0     0.00       0.00     0.00        0.00     0.00     0.00   
5     0.03       0.04     0.07        0.24     3.16     0.13   
2     0.16       0.00     0.14        4.36     0.00     0.00   
7    12.27       0.00     0.01        0.03    10.74     0.00   
1     0.86       0.00     0.00        0.62     0.00     0.00   

   Bosnia-Herzegovina  Brasil  Bulgaria  Burkinafaso  Canada  Chile  China  \
0                0.00    0.00     49.24         0.00    0.00   0.00  39.17   
5                0.00    0.01      0.08         0.00    0.33   0.00   4.30   
2                0.00    0.00      0.00         0.00    0.00   0.00   0.00   
7                0.00    0.00      0.04         0.00    0.02   0.00   0.35   
1                0.00    0.00      1.19         0.00    0.00   0.00   0.00   

   Congo  Cyprus  Czech Republic  Côte d'Ivoire  Denmark  Egypt  Estonia  \
0   0.00    0.00            0.00           0.00     0.00   0.00     0.00   
5   0.01    0.03            0.11           0.00     0.03   0.00     0.66   
2   0.00    0.08            0.00           0.00     0.00   1.53     0.00   
7   0.01    0.00            0.11           0.00     0.00   0.00     0.13   
1   0.00    0.00            0.00           0.00     0.00   0.00     0.00   

   Finland  France  Gana  Georgia  Germany  Great Britain  Greece  Guinea  \
0     0.00    0.00  0.00     0.00     0.00           0.00    0.00    0.00   
5     0.04    0.22  0.00     0.00     1.84           0.53    0.01    0.00   
2     0.00    1.59  0.00     0.00     0.09           0.12    0.00    0.00   
7     0.00   11.18  0.00     0.87     0.09           0.02    0.00    0.00   
1     0.00    0.00  4.54     0.00     0.00           0.00    0.00    0.00   

   Hong Kong  Hungary  Iceland  India  Iran  Iraq  Ireland  Israel  Italy  \
0       0.00     0.00     0.00   0.00  0.00  0.00     0.00    0.00   0.00   
5       0.04     0.01     0.00   0.00  0.00  0.00     0.00    0.37   0.07   
2       0.00     0.00     0.00   0.00  0.25  0.33     0.00    0.00   0.22   
7       0.01     0.00     0.00   0.00  0.00  0.00     0.00    0.32   0.12   
1       0.00     0.00     0.00   4.69  0.00  0.00     0.00    0.48   0.82   

   Japan  Jordan  Kazakhstan  Kenya  Kuwait  Kyrgyzstan  Laos  Latvia  \
0   0.00    0.00        0.00   0.00    0.00        0.00  0.00    0.00   
5   0.53    0.00        3.57   0.00    0.00        0.13  0.06    1.56   
2   0.01    0.00        0.00   0.00    0.00        0.00  0.00    0.00   
7   0.00    0.00        2.14   0.00    0.00        0.01  0.00    0.93   
1   0.00    0.00        0.00   0.00    0.00        0.00  0.00    0.00   

   Lebanon  Liberia  Libia  Liechtenstein  Lithuania  Luxembourg  Macedonia  \
0     0.00     0.00   0.00           0.00       0.00        0.00       0.00   
5     0.00     0.00   0.00           0.00       0.88        0.01       0.00   
2     0.00     0.00   0.00           0.02       0.00        0.00       0.00   
7     0.00     0.00   0.00           0.00       0.70        0.00       0.00   
1     0.00     0.00   0.00           0.00       3.45        0.00       0.00   

   Mala  Malaysia  Mauritania  Mexico  Moldova  Mongolia  Montenegro  Morocco  \
0  0.00      0.00        0.00    0.00     0.00      0.00        0.00     0.00   
5  0.00      0.00        0.00    0.00     0.06      0.25        0.00     0.00   
2  0.00      0.00        0.00    0.00     0.10      0.00        0.00     0.00   
7  0.00      0.00        0.00    0.00     0.00

In [50]:
# The 1st most exported product category.
country_share_exp.iloc[:1,:].replace({0:np.nan}).dropna(axis=1)

Country Category  Bulgaria  Canada  China  Finland  Spain  russia        Total
0           2603     49.24    0.00  39.17     0.00  11.59    0.00 487347726.51

Of Copper ores and concentrates *(p2603)*, **49.24%** was exported to Bulgaria, while **39.17%** went to China, and **11.59%** to Spain.

In [51]:
# The second most exported product category.
country_share_exp.iloc[1:2,:].replace({0:np.nan}).dropna(axis=1)

Country Category  Armenia  Australia  Austria  Azerbaijan  Belarus  Belgium  \
1           2204     0.03       0.04     0.07        0.24     3.16     0.13   

Country  Brasil  Bulgaria  Canada  China  Congo  Cyprus  Czech Republic  \
1          0.01      0.08    0.33   4.30   0.01    0.03            0.11   

Country  Denmark  Estonia  Finland  France  Germany  Great Britain  Greece  \
1           0.03     0.66     0.04    0.22     1.84           0.53    0.01   

Country  Hong Kong  Hungary  Iceland  India  Ireland  Israel  Italy  Japan  \
1             0.04     0.01     0.00   0.00     0.00    0.37   0.07   0.53   

Country  Kazakhstan  Kenya  Kyrgyzstan  Laos  Latvia  Lithuania  Luxembourg  \
1              3.57   0.00        0.13  0.06    1.56       0.88        0.01   

Country  Mala  Malaysia  Mexico  Moldova  Mongolia  Netherlands  Niger  \
1        0.00      0.00    0.00     0.06      0.25         0.14   0.01   

Country  Nigeria  Norway  Poland  Portugal  Qatar  Romania  Singapore  \
1           0.01    0.20    5.57      0.01   0.00     0.03       0.02   

Country  Slovakia  Slovenia  South Korea  Spain  Sweden  Switzerland  Taiwan  \
1            0.01      0.01         0.12   0.05    0.21         0.03    0.02   

Country  Tajikista  Thailand  Turkmenistan  Türkiye  UA Emirates  Uganda  \
1             0.10      0.02          0.14     0.34         0.10    0.00   

Country  Ukraine  United States  Uzbekistan  russia        Total  
1           5.52           2.03        0.16   65.77 239916256.74

Regarding **Wine** *(p2204)*, **65.77%** was exported to russia, **5.57%** to Poland, **5.52%** to Ukraine, **2.03%** to the United States, **4.30%** to China, and **3.57%** to Kazakhstan.